### PR TITLE
Added a recipe for `gams-ac' package.

### PR DESCRIPTION
**Pull request by ShiroTakeda**
### Brief summary of what the package does

This package provides auto-complete source for gams-mode ([https://github.com/ShiroTakeda/gams-mode](https://github.com/ShiroTakeda/gams-mode)).

### Direct link to the package repository

https://github.com/ShiroTakeda/gams-ac

### Your association with the package

I am the author and maintainer.

### Relevant communications with the upstream package maintainer

None needed

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses). 
- [x] I've read [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: recipes/gams-ac [hunk 1]
```diff
@@ -0,0 +1,5 @@
+(gams-ac
+ :repo "ShiroTakeda/gams-ac"
+ :fetcher github
+ :files ("gams-ac.el")
```
Comment (DamienCassou) [inline recipes/gams-ac]: please remove the `:files` line, it is [probably useless](https://github.com/melpa/melpa#recipe-format).

**Comment by DamienCassou** [comment]
Some feedback:

- you might want to add `-*- lexical-binding: t; -*-` at the end of the first line
- please remove the `.` at the end of the last line
- the [the emacs lisp style guide](https://github.com/bbatsov/emacs-lisp-style-guide) recommends

    > [Placing] all trailing parentheses on a single line instead of distinct lines

- I'm not sure your macro makes sense because the resulting code is not much more complex. Additionally, the variables could benefit from a docstring that the macro doesn't support. I recommend inlining the macro-expansion and removing the macro-definition
- I believe `gams-ac-attach-$` is not worth it: I would inline it

**Comment by ShiroTakeda** [comment]
Dear Damien

Thank you very much for useful suggestions and advises.

I want to modify the code.

**Comment by ShiroTakeda** [comment]
Dear Damien

According to your suggestions, I rewrote the entire part of `gams-ac.el`.

I abolished macro and use normal variables and functions instead.

**Comment by purcell** [comment]
Thanks - I think you no longer need to `(require 'cl)`, and you might like to add an `;;;###autoload` cookie comment for the `gams-ac-setup` and `gams-ac-after-init-setup` functions. In any case, this is ready to merge. :-)

**Comment by ShiroTakeda** [comment]
Thank you, Steve.

According to your suggestions, I rewrote the code.

**Pull request merged**